sdk/core/azure-core/src/main/java/com/azure/core/util/Configuration.java

```diff
@@ -214,18 +215,57 @@
     @SuppressWarnings("StaticInitializerReferencesSubClass")
     public static final Configuration NONE = new NoopConfiguration();
 
+    /*
+     * Gets the global configuration shared by all client libraries.
+     */
+    private static final Configuration GLOBAL_CONFIGURATION = new Configuration();
+
     private final ConcurrentMap<String, String> configurations;
+    private final ConfigurationSource[] configurationSources;
+    private final String sectionName;
+    private final Configuration base;
 
     /**
      * Constructs a configuration containing the known Azure properties constants.
      */
     public Configuration() {
         this.configurations = new ConcurrentHashMap<>();
+        this.configurationSources = CONFIG_SOURCES;
+        this.sectionName = null;
+        this.base = null;
         loadBaseConfiguration(this);
     }
 
-    private Configuration(ConcurrentMap<String, String> configurations) {
+    public Configuration(ConfigurationSource... additionalSources) {
+        this.configurations = new ConcurrentHashMap<>();
+
+        // TODO(configuration): priority, should we add env var sources by default
```

Comment:
I'd actually lean towards this constructor only using the `additionalSources`, I'd treat it as the "I know what I'm doing and what I want" scenario. There are times where it may be beneficial to ignore the environment variables, system properties, or both.

Reply:
I agree it makes sense, I think we're in the gray area here as guidelines tell us to load env vars no matter what
https://azure.github.io/azure-sdk/general_azurecore.html#environment-variables

It would also be the least surprise to users if we keep loading them